modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -232,6 +232,51 @@ public Object addSystemObjectRelationshipMappingTableValues(
 			systemObjectDefinitionManager);
 	}
 
+	@Override
+	public ObjectEntry copyObjectEntryByVersion(
+			DTOConverterContext dtoConverterContext,
+			ObjectDefinition objectDefinition, long objectEntryId, int version)
+		throws Exception {
+
+		ObjectEntry objectEntry = getObjectEntryByVersion(
+			dtoConverterContext, objectEntryId, version);
+
+		objectEntry.setExternalReferenceCode(() -> null);
+		objectEntry.setId(() -> null);
+
+		_removeReadOnlyProperties(objectDefinition, objectEntry);
+
+		return addObjectEntry(
+			dtoConverterContext, objectDefinition, objectEntry,
+			objectEntry.getScopeKey());
+	}
+
+	@Override
+	public ObjectEntry copyObjectEntryByVersion(
+			DTOConverterContext dtoConverterContext,
+			String externalReferenceCode, ObjectDefinition objectDefinition,
+			int version)
+		throws Exception {
+
+		com.liferay.object.model.ObjectEntry serviceBuilderObjectEntry =
+			_objectEntryService.getObjectEntry(
+				externalReferenceCode, objectDefinition.getCompanyId(),
+				getGroupId(objectDefinition, null));
+
+		ObjectEntry objectEntry = getObjectEntryByVersion(
+			dtoConverterContext, serviceBuilderObjectEntry.getObjectEntryId(),
+			version);
+
+		objectEntry.setExternalReferenceCode(() -> null);
```

Comment:
Can you create a private method `_copyVersionedObjectEntry()` to reuse this logic?

```
		objectEntry.setExternalReferenceCode(() -> null);
		objectEntry.setId(() -> null);

		_removeReadOnlyProperties(objectDefinition, objectEntry);

		return addObjectEntry(
			dtoConverterContext, objectDefinition, objectEntry,
			objectEntry.getScopeKey());
```